lib/rpmsg/rpmsg_virtio.c

```diff
@@ -637,7 +645,7 @@ int rpmsg_init_vdev(struct rpmsg_virtio_device *rvdev,
 	 * Create name service announcement endpoint if device supports name
 	 * service announcement feature.
 	 */
-	if (rdev->support_ns) {
+	if (rdev->support_ns || rdev->support_ack) {
```

Comment:
"ack" is better not to sent with NS ept, instead, it should sent from the locally created ept to indicate it is ready.

Reply:
You mean that "ack" send through the locally created ept?
How to avoid the ack message id conflict with the message defined by protocal designer?

Reply:
> You mean that "ack" send through the locally created ept?
> How to avoid the ack message id conflict with the message defined by protocal designer?

RPMsg requires communication between endpoints, my understanding for the ack means it is ready to have talk to the otherside, that is local endpoint has been created. The ack will finish the endpoints binding. Agree that there is concern that how to define the message ID.
I am fine to reserved an ACK message ID, but we need to agree with the community.
@arnop2, @galak, @andersson, what's your thoughts on this?

Reply:
I will prepare a patch for Linux kernel by end of day for discussion.

Reply:
@arnop2, @galak, @andersson and @wjliang, here is the patch for kernel side:
https://patchwork.kernel.org/patch/10807413/